library/src/main/scala/scala/scalajs/runtime/RuntimeLong.scala

```diff
@@ -304,7 +304,29 @@ final class RuntimeLong(val lo: Int, val hi: Int)
   def *(b: RuntimeLong): RuntimeLong = {
     val alo = a.lo
     val blo = b.lo
-    new RuntimeLong(alo * blo, RuntimeLong.timesHi(alo, a.hi, blo, b.hi))
+
+    val a0 = alo & 0xffff
+    val a1 = alo >>> 16
+    val b0 = blo & 0xffff
+    val b1 = blo >>> 16
+
+    val a0b0 = a0 * b0
+    val a1b0 = a1 * b0 // collapses to 0 when a is constant and 0 <= a <= 0xffff
```

Comment:
Ah, but that is (almost) never going to happen. If `b` is constant, then either `a` is constant too, and the multiplication is constant-folded away; or `a` is not constant, and the optimizer would have normalized the multiplication as `b * a` so that the lhs is constant.

The only case where `a0b1` would be constant 0 is if `b`'s `lo` part is constant but not its `hi` part. That's not a likely scenario, though (not seen at all in our test suite).